lib/quiz/card_list.dart

```diff
@@ -0,0 +1,303 @@
+  final Map<String, dynamic> input;
+  final OptionCategory optionsType;
+  final Function onEnd;
+
+  const CardList(
+      {Key key,
+      this.input = testMap,
+      this.optionsType = OptionCategory.many,
+      this.onEnd})
+      : super(key: key);
+
+  @override
+  State createState() => new CardListState();
+}
+
+class CardListState extends State<CardList> {
+  List<String> clicked = [],
+      choice = [],
+      clickedChoices = [],
+      shuffledChoices = [];
+  List<bool> rightOrWrong = [];
+  bool displayIcon = false;
+  int count = 0, correctChoices = 0;
+
+  @override
+  void initState() {
+    super.initState();
+    print("Card List Initstate");
+    _initBoard();
+  }
+
+  void _initBoard() {
+    // Adding data of choices given by parent class to the choices and shuffledchoices variable
+    if (widget.optionsType == OptionCategory.many) {
+      for (int i = 0; i < widget.input['answer'].length; i++) {
+        choice.add(widget.input['answer'].cast<String>()[i]);
+        shuffledChoices.add(widget.input['answer'].cast<String>()[i]);
+      }
+      if (widget.input['choices'] != null) {
+        for (int i = 0; i < widget.input['choices'].length; i++) {
+          choice.add(widget.input['choices'].cast<String>()[i]);
+          shuffledChoices.add(widget.input['choices'].cast<String>()[i]);
+        }
+      }
+    } else if (widget.optionsType == OptionCategory.oneAtATime) {
+      choice = shuffledChoices = widget.input['answer'];
+      for (int i = 0; i < widget.input['choices'].length; i++) {
+        choice.add(widget.input['choices']);
+        shuffledChoices.add(widget.input['choices']);
+      }
+    } else if (widget.optionsType == OptionCategory.pair) {
+      for (int i = 0; i < widget.input['choices'].length; i++) {
+        choice.add(widget.input['choices'].cast<String>()[i]);
+        shuffledChoices.add(widget.input['choices'].cast<String>()[i]);
+      }
+    }
+
+    // Shuffling choices
+    shuffledChoices.shuffle();
+
+    // Array to keep track if a choice is clicked or not
+    clicked = choice.map((e) => "no").toList(growable: false);
+
+    // Array to keep track if choice is correct or incorrect
+    if (widget.input['correct'] == null) {
+      rightOrWrong = choice.map((i) => false).toList(growable: false);
+    } else {
+      for (var i = 0; i < widget.input['choicesRightOrWrong'].length; i++) {
+        rightOrWrong.add(widget.input['choicesRightOrWrong'][i]);
+      }
+    }
+    print("Value of RightOrWrong Array after getting values - $rightOrWrong");
+  }
+
+  Widget _buildItem(int index, String text, int k, double ht) {
+    // Universal Button for mapping keys, text/image to be shown, Button's present status and a function to perform desired actions
+    return new Container(
+        height: ht / 8.0,
+        child: QuizButton(
+            key: new ValueKey<int>(index),
+            text: text,
+            buttonStatus: (widget.input['correct'] == null
```

Comment:
Are you using widget.input["correct"] == null to check if we are in quiz mode or answer mode? If so, it is better to check it in the beginning and use it as a mode. Will clean up the code